### PR TITLE
Ignore CI triggers for changes on the other one

### PR DESCRIPTION
**Pull request by ylavoie**
Make sure that CI will ignore changes done on the other CI configuration

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -13,6 +13,14 @@ aliases:
 
 # Reuseable commands
 commands:
+
+  check-changed-files-or-halt:
+    parameters:
+      pattern:
+        type: string
+    steps:
+      - run: git show -m HEAD --name-only --pretty="" | egrep -q -v '<< parameters.pattern >>' || circleci step halt
```
Comment (ehuelsmann) [inline .circleci/config.yml:22]: Will this cancel the run if there is no file changed with the content `<< parameters.pattern >>` ? If so, I don't think this is a correct condition: that'll be the case in most situations.

Maybe a better condition is: filter out all files in `.github` and check if there are any files left which have changed?

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -8,9 +8,13 @@ on:
   push:
     branches-ignore:
       - 'master'
+    paths-ignore:
+      - '.circleci'
```
Comment (ehuelsmann) [inline .github/workflows/main.yml:12]: This feels like a much more solid condition indeed!

File: .circleci/config.yml (hunk 1 above)
Comment (ylavoie) [reply .circleci/config.yml:22]: - git show list all files involved in the last PR
- grep -q -v set its exit status to success if there are no files matching `<< parameters.pattern >>` and the step will halt

Comment (ehuelsmann) [reply .circleci/config.yml:22]: Right. And the pattern you're using is `^.github/`; so, if no files match that, the step will halt. But when I only change one of the workflow scripts (e.g. `lib/LedgerSMB/Scripts/setup.pm`) then the build will be cancelled for not matching `.github/`, right?

Comment (ylavoie) [reply .circleci/config.yml:22]: The build is cancelled only if .github is the only thing in the modified files.

Comment (ehuelsmann) [reply .circleci/config.yml:22]: ah! sorry about misunderstanding your code! I see I overlooked the `-v` switch's effects.

**Review by ehuelsmann — APPROVED** [review]

**Pull request merged**